### ISSUE TITLE
Fish completion for rpm-ostree is broken

### ISSUE THREAD

**Issue by yuntaz2** [fish-shell/fish-shell#9667]
**YOUR TEXT HERE**
`fish --version`
fish, version 3.6.0

`uname -a`
Linux 6.1.18-200.fc37.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Mar 11 16:09:14 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

`echo $TERM`
xterm-256color

Third party extension does not affect this issue.

To reproduce the issue, please install Fish shell on Fedora 37 Silverblue or Kinoite, which use rpm-ostree as their hybrid image/package system. By default, there is no tab completion available for rpm-ostree. Running `fish_update_completion` should automatically generate the ~/.local/share/fish/generated_completions/rpm-ostree.fish file with the completion options. However, it seems that the completion generated is not correct.

### Autogenerated from man page /usr/share/man/man1/rpm-ostree.1.gz ###
```
complete -c rpm-ostree -l format -d 'option uses'
complete -c rpm-ostree -l reboot -d or
complete -c rpm-ostree -s r -d 'to initiate a reboot after the upgrade is prepared'
complete -c rpm-ostree -l preview -d 'download enough metadata to inspect the RPM diff, but do not actually create …'
complete -c rpm-ostree -l cache-only -d or
complete -c rpm-ostree -s C -d 'to perform the operation without trying to download the target tree from the …'
complete -c rpm-ostree -l download-only -d 'to only download the target ostree and layered RPMs without actually performi…'
complete -c rpm-ostree -l allow-inactive -d 'option is provided'
complete -c rpm-ostree -l idempotent -d 'to do nothing if a package request is already present instead of erroring out'
complete -c rpm-ostree -l dry-run -d or
complete -c rpm-ostree -s n -d 'to exit after printing the transaction rather than downloading the packages a…'
complete -c rpm-ostree -l apply-live -d 'will perform a subsequent apply-live operation to apply changes to the booted…'
complete -c rpm-ostree -l branch -d or
complete -c rpm-ostree -l remote -d 'options mentioned below'
complete -c rpm-ostree -s b -d 'to pick a branch name'
complete -c rpm-ostree -s m -d 'to pick a remote name'
complete -c rpm-ostree -l allow-downgrade -d 'to permit deployment of chronologically older trees'
complete -c rpm-ostree -l check -d 'to just check if an upgrade is available, without downloading it or performin…'
complete -c rpm-ostree -l append
complete -c rpm-ostree -l delete
complete -c rpm-ostree -l delete-if-present
complete -c rpm-ostree -l replace
complete -c rpm-ostree -l deploy-index -d or
complete -c rpm-ostree -l import-proc-cmdline -d 'to instead base them off of a specific deployment or the current boot'
complete -c rpm-ostree -o p/--pending -d 'to remove the pending deployment, and'
complete -c rpm-ostree -o r/--rollback -d 'to remove the rollback'
complete -c rpm-ostree -o b/--base -d 'option does not affect finished deployments, but will clean up any transient …'
complete -c rpm-ostree -o m/--repomd -d 'option cleans up cached RPM repodata and any partially downloaded (but not im…'
complete -c rpm-ostree -l enable -d 'to turn on client side initramfs regeneration'
complete -c rpm-ostree -l arg
complete -c rpm-ostree -s I
complete -c rpm-ostree -l disable -d 'option will disable regeneration'
complete -c rpm-ostree -l track -d 'to start tracking a specific file.  Can be specified multiple times'
complete -c rpm-ostree -l untrack -d or
complete -c rpm-ostree -l untrack-all -d 'to stop tracking files'
complete -c rpm-ostree -l force-sync -d 'to simply generate a new deployment with the latest versions of tracked files…'
complete -c rpm-ostree -l reset -d 'to reset the filesystem tree to the booted commit'
complete -c rpm-ostree -l target -d 'may be used to target an arbitrary OSTree commit'
complete -c rpm-ostree -l allow-replacement -d 'enables live updates and removals for existing packages'
```

This completion is incorrect, probably because the man page of rpm-ostree is not in the correct format.
The correct result should be something like:

```
# Define subcommands for rpm-ostree
set -l subcommands apply-live compose cancel cleanup db deploy initramfs initramfs-etc install kargs override refresh-md reload rebase reset rollback status uninstall upgrade usroverlay

# File completions also need to be disabled
complete -c rpm-ostree -f

# Define auto-completion options for rpm-ostree
complete -c rpm-ostree -n "not __fish_seen_subcommand_from $subcommands" -a "$subcommands"

# deploy
complete -c rpm-ostree -n '__fish_seen_subcommand_from deploy' -l unchanged-exit-77 -d 'Exit status 77 to indicate that the system is already on the specified commit'
complete -c rpm-ostree -n '__fish_seen_subcommand_from deploy' -s r -l reboot -d 'Initiate a reboot after the upgrade is prepared'
complete -c rpm-ostree -n '__fish_seen_subcommand_from deploy' -l preview -d 'Download enough metadata to inspect the RPM diff, but do not actually create a new deployment'
complete -c rpm-ostree -n '__fish_seen_subcommand_from deploy' -s C -l cache-only -d 'Perform the operation without trying to download the target tree from the remote nor the latest packages'
complete -c rpm-ostree -n '__fish_seen_subcommand_from deploy' -l download-only -d 'Only download the target ostree and layered RPMs without actually performing the deployment. This can be used with a subsequent --cache-only invocation to perform the operation completely offline.'

# install
complete -c rpm-ostree -n '__fish_seen_subcommand_from install' -l idempotent -d 'Do nothing if a package request is already present instead of erroring out'
complete -c rpm-ostree -n '__fish_seen_subcommand_from install' -s r -l reboot -d 'Initiate a reboot after the deployment is prepared'
complete -c rpm-ostree -n '__fish_seen_subcommand_from install' -s n -l dry-run -d 'Exit after printing the transaction rather than downloading the packages and creating a new deployment'
complete -c rpm-ostree -n '__fish_seen_subcommand_from install' -l allow-inactive -d 'Allow requests for packages that are already in the base layer'
complete -c rpm-ostree -n '__fish_seen_subcommand_from install' -s C -l cache-only -d 'Perform the operation without trying to download the latest packages'
complete -c rpm-ostree -n '__fish_seen_subcommand_from install' -l download-only -d 'Only download the target layered RPMs without actually performing the deployment. This can be used with a subsequent --cache-only invocation to perform the operation completely offline.'
complete -c rpm-ostree -n '__fish_seen_subcommand_from install' -l apply-live -d 'Perform a subsequent apply-live operation to apply changes to the booted deployment'

# uninstall
complete -c rpm-ostree -n '__fish_seen_subcommand_from uninstall' -s r -l reboot -d 'Initiate a reboot after the deployment is prepared'
complete -c rpm-ostree -n '__fish_seen_subcommand_from uninstall' -s n -l dry-run -d 'Exit after printing the transaction rather than downloading the packages and creating a new deployment'

# rebase
complete -c rpm-ostree -n '__fish_seen_subcommand_from rebase' -l branch -d 'Pick a branch name'
complete -c rpm-ostree -n '__fish_seen_subcommand_from rebase' -l remote -d 'Pick a remote name'
complete -c rpm-ostree -n '__fish_seen_subcommand_from rebase' -s C -l cache-only -d 'Perform the rebase without trying to download the target tree from the remote nor the latest packages'
complete -c rpm-ostree -n '__fish_seen_subcommand_from rebase' -l download-only -d 'Only download the target ostree and layered RPMs without actually performing the deployment. This can be used with a subsequent --cache-only invocation to perform the operation completely offline'

# rollback
complete -c rpm-ostree -n '__fish_seen_subcommand_from rollback' -s r -l reboot -d 'Initiate a reboot after rollback is prepared'

# upgrade
complete -c rpm-ostree -n '__fish_seen_subcommand_from upgrade' -l allow-downgrade -d 'Permit deployment of chronologically older trees'
complete -c rpm-ostree -n '__fish_seen_subcommand_from upgrade' -l preview -d 'Download only /usr/share/rpm in order to do a package-level diff between the two versions'
complete -c rpm-ostree -n '__fish_seen_subcommand_from upgrade' -l check -d 'Just check if an upgrade is available, without downloading it or performing a package-level diff. Using this flag will force an update of the RPM metadata from the enabled repos in /etc/yum.repos.d/, if there are any layered packages'
complete -c rpm-ostree -n '__fish_seen_subcommand_from upgrade' -s C -l cache-only -d 'Perform the upgrade without trying to download the latest tree from the remote nor the latest packages'
complete -c rpm-ostree -n '__fish_seen_subcommand_from upgrade' -l download-only -d 'Only download the target ostree and layered RPMs without actually performing the deployment. This can be used with a subsequent --cache-only invocation to perform the operation completely offline'
complete -c rpm-ostree -n '__fish_seen_subcommand_from upgrade' -s r -l reboot -d 'Initiate a reboot after the upgrade is prepared'
complete -c rpm-ostree -n '__fish_seen_subcommand_from upgrade' -l unchanged-exit-77 -d 'Exit status 77 to indicate that the system is already up to date. This tristate return model is intended to support idempotency-oriented systems automation tools like Ansible.'

# kargs
complete -c rpm-ostree -n '__fish_seen_subcommand_from kargs' -l editor -d 'Use an editor to modify the kernel arguments'
complete -c rpm-ostree -n '__fish_seen_subcommand_from kargs' -l append -d 'Append a kernel argument'
complete -c rpm-ostree -n '__fish_seen_subcommand_from kargs' -l append-if-missing -d 'Append a kernel argument if it is not present'
complete -c rpm-ostree -n '__fish_seen_subcommand_from kargs' -l delete -d 'Delete a kernel argument'
complete -c rpm-ostree -n '__fish_seen_subcommand_from kargs' -l delete-if-present -d 'Delete a kernel argument if it is already present'
complete -c rpm-ostree -n '__fish_seen_subcommand_from kargs' -l replace -d 'Replace an existing kernel argument'
complete -c rpm-ostree -n '__fish_seen_subcommand_from kargs' -l unchanged-exit-77 -d 'Exit status 77 to indicate that the kernel arguments have not changed'
complete -c rpm-ostree -n '__fish_seen_subcommand_from kargs' -l deploy-index -d 'Use the specified deployment index to modify kernel arguments'
complete -c rpm-ostree -n '__fish_seen_subcommand_from kargs' -l import-proc-cmdline -d 'Use the current boot kernel arguments to modify the kernel arguments'

# cleanup
complete -c rpm-ostree -n '__fish_seen_subcommand_from cleanup' -s p -l pending -d 'Remove the pending deployment'
complete -c rpm-ostree -n '__fish_seen_subcommand_from cleanup' -s r -l rollback -d 'Remove the default rollback deployment'
complete -c rpm-ostree -n '__fish_seen_subcommand_from cleanup' -s b -l base -d 'Remove any transient allocated space from interrupted operations'
complete -c rpm-ostree -n '__fish_seen_subcommand_from cleanup' -s m -l repomd -d 'Clean up cached RPM repodata and partially downloaded packages'

# initramfs
complete -c rpm-ostree -n '__fish_seen_subcommand_from initramfs' -l enable -d 'Turn on client side initramfs regeneration'
complete -c rpm-ostree -n '__fish_seen_subcommand_from initramfs' -l arg -d 'Append additional custom arguments to the initramfs program'
complete -c rpm-ostree -n '__fish_seen_subcommand_from initramfs' -l disable -d 'Disable initramfs regeneration'

# initramfs-etc
complete -c rpm-ostree -n '__fish_seen_subcommand_from initramfs-etc' -l track -d 'Start tracking a specific file. Can be specified multiple times.'
complete -c rpm-ostree -n '__fish_seen_subcommand_from initramfs-etc' -l untrack -d 'Stop tracking files.'
complete -c rpm-ostree -n '__fish_seen_subcommand_from initramfs-etc' -l untrack-all -d 'Stop tracking all files.'
complete -c rpm-ostree -n '__fish_seen_subcommand_from initramfs-etc' -l force-sync -d 'Generate a new deployment with the latest versions of tracked files without upgrading.'

# apply-live
complete -c rpm-ostree -n '__fish_seen_subcommand_from apply-live' -l reset -d 'Reset the filesystem tree to the booted commit'
complete -c rpm-ostree -n '__fish_seen_subcommand_from apply-live' -l target -d 'Target an arbitrary OSTree commit'
complete -c rpm-ostree -n '__fish_seen_subcommand_from apply-live' -l allow-replacement -d 'Enable live updates and removals for existing packages'
```
We would appreciate it if someone could provide better tab completion for rpm-ostree. Thank you!

**Comment by faho** [comment]
Quite simply: "Patches welcome".

Have you thought of just opening a PR with what you have posted here?

**Comment by mqudsi** [comment]
I completed the changes needed for this and merged #9669 targeted at 3.6.2/3.7.0
